src/presentations/Header.js

```diff
@@ -0,0 +1,124 @@
+// import CartIcon from '../../static/images/cart.svg';//'';
+import LazyLoad from 'react-lazyload';
+class Header extends React.Component {
+     constructor(props) {
+        super(props);
+        this.state = {
+            logo: '',
+            clickable: false,
+            userName:'',
+            itemCount: 0,
+            dimensions:{}
+        }
+        // this.clickEvent = this.clickEvent.bind(this);
+        // this.showHome = this.showHome.bind(this);
+        // this.doResizeActions = this.doResizeActions.bind(this);
+        // this.cartClick = this.cartClick.bind(this);
+        // this.resetCart = this.resetCart.bind(this);
+        // this.resize = this.resize.bind(this);
+    }
+    componentDidMount(){
+        if(this.props.signin.userData && this.props.signin.userData.userEmail){
+            this.setState({userName:this.props.signin.userData.userEmail});
+        } else if(this.props.signup && this.props.signup.userData && this.props.signup.userData.userEmail){
+            this.setState({userName:this.props.signup.userData.userEmail});
+        }
+        this.doResizeActions();
+        window.addEventListener('resize',this.resize,false);
+    }
+    componentWillUnmount(){
+        window.removeEventListener('resize',this.resize,false);
+    }
+
+    static getDerivedStateFromProps(props,state){
+        if(props.cartInfo){
+            state.itemCount= props.cartInfo.length;
+        }
+        if(props.productInfo && props.productInfo.cartItems){
+            state.itemCount= props.productInfo.cartItems.length
+        }
+        return state;
+    }
+    resize=()=>{
+        this.doResizeActions();
+    }
+    resetCart=()=>{
+        this.props.resetCartData();
+    }
+    doResizeActions=()=>{
+        let dimensions = getViewPortDimensions();
+        this.setState({dimensions: dimensions});
+    }
+    cartClick=()=>{
+        if(this.state.dimensions && this.state.dimensions.width && this.state.dimensions.width < 480){ //1024
+            this.props.router.push('/cart');
+        } else {
+            this.props.cartClick();
+        }
+    }
+    clickEvent=(event)=>{}
+    showHome=()=>{
+        this.props.router.push('/');
+    }
+    render(){   
+        const {userName,itemCount} = this.state;
+        return (<div className="header-area">
+            <header>
+                <nav className="sticky-nav">
+                    <div className="row">
+                        <img onClick={this.showHome} src={window.location.origin + '/static/images/logo.png'} alt="logo" className="logo" /> {/* {Logo} */}
+                        <ul className="main-nav js--main-nav">
+                            <li><Link to={`/home`} >Home</Link></li>
+                            <li><Link to="/products">Products</Link></li>
+                            <div className="logo-icon-header">
+                            <li><Link to={`/home`} ><i className="ion-ios-home"></i></Link></li>
```

Comment:
div as a direct descendent of ul and a sibling of li?

Reply:
Corrected.